### ISSUE TITLE
[Cronjob] possible Vuln.

### ISSUE THREAD

**Issue by Bizarrus** [fruithost/Modules#4]
Cronjobs will be runned over Daemon and will be executed as `root`. Prevent these, to exeute the Cronjob-Entrys as `www-data`

**Comment by Bizarrus** [comment]
# Further information
The [`Daemon`](https://github.com/fruithost/Binary) must run with root rights, as server services must also be restarted or extended write rights to system files are required, among other things.

# Possibilities
Currently, module files ([deinstall.php](https://github.com/fruithost/Binary/blob/master/cli.php#L743), [install.php](https://github.com/fruithost/Binary/blob/master/cli.php#L427) and [daemon.php](https://github.com/fruithost/Binary/blob/master/cli.php#767)) are directly included in the [`Daemon`](https://github.com/fruithost/Binary) using `require_one` and thus automatically receive executable root rights.

Here it would be possible to call and execute these via an `FPM` spawn process. The mechanics for this already exist in [PHP.class.php](https://github.com/fruithost/Panel/blob/master/classes/PHP.class.php#L42) and would only need to be adapted and ported.

However, as there is also the possibility that modules may require root rights of the system (especially in the installation process), the corresponding module files should have optional settings that have been optionally set by the module developer.

Here it would be possible to add the following value at the beginning of the files:

**install.php**
```php
#!fruithost:permissions:root
<?php
    // The install script
?>
```

However, as there is also the possibility that modules may require root rights of the system (especially in the installation process), the corresponding module files should have optional settings that have been optionally set by the module developer.

Here it would be possible to add the following value at the beginning of the files:

**deinstall.php**
```php
#!fruithost:permissions:root
<?php
    // The deinstall script
?>
```

**daemon.php**
```php
#!fruithost:permissions:root
<?php
    // The daemon script
?>
```

By specifying `#!fruithost:permissions:root` at the beginning of the file, the daemon is instructed to include and execute the corresponding files normally using `require_once`. If the specification does not exist, the `FPM` spawn process is used for execution.

**Comment by Bizarrus** [comment]
Fixed in [#5c17ef8](https://github.com/fruithost/Binary/commit/5c17ef81198bfea7e40442bba4dff23b4488cd80). Check out the documentation: [/Documentation/blob/main/Modules/Module%20Permissions.md](https://github.com/fruithost/Documentation/blob/main/Modules/Module%20Permissions.md)
